### ISSUE TITLE
unable to find device on smartthings during configuration

### ISSUE THREAD

**Issue by marcoevcom** [ollo69/ha-samsungtv-smart#15]
Hello, I've tried to set up my Samsung TV following your guide, everything went as it was supposed to, but when I try to add the device to my home assistant integrations page, I get the error saying that my device was not found.

One thing I noticed during setup was that the address you provide to list our smartthings devices: https://graph-eu01-euwest1.api.smartthings.com/device/list does not list devices, and this one: https://graph-na04-useast2.api.smartthings.com/device/list does list.

maybe this is just a regionalization thing from smartthings but it is getting in the way of this nice integration to work.

I'm available to help test anything you need.

Thanks in advance

**Comment by sn3kyJ3di** [comment]
I tried setting mine up as well and get the error saying it wasn't found.  I used the API key and device ID.  No difference.

**Comment by ollo69** [comment]
I will investigate. 
As I know SmartThings should not be region related, but I try to login to https://graph-na04-useast2.api.smartthings.com/device/list  and my TV is not in the list.
This is new also  for me....

**Comment by ollo69** [comment]
I think that this is issue is related to the generated token, because the smartthings url that I use to access smartthings API have no region settings.
Are you using "samsung account" or "smartthings account"?
Try to generate new token using samsung account (create a new one if you don't have) and let me know if this work.

**Comment by ollo69** [comment]
I close this issue, pls re-open if you have additional comment or information.

**Comment by marcoevcom** [comment]
Sorry to answer this after all this time.
I used my samsung account to create the token.
![samsungerror](https://user-images.githubusercontent.com/12731842/82932538-f7b81500-9f5e-11ea-94da-7e812c8c36c5.png)

**Comment by ollo69** [comment]
I think that this is a issue of SmartThings API, because there are nothing related to the region inside documentation so there is nothing that I can do.
My only suggestion is to try de-configure everything, that  means delete TV from SmartThings phone apps and add again, and then generate a new token (be sure to set correct permission).

**Comment by marcoevcom** [comment]
hi ollo69,  I really couldn't setup using the wizard on the features page. But I dedicated more time to this and decided to do a manual setup. I worked !!  I used my api_key and the TV's IP address.
It is working very well. thanks for this integration.

**Comment by ollo69** [comment]
This is strange behavior, so API is working but fail to detect device in configuration? I open again this ticket, need further investigation.

**Comment by ollo69** [comment]
Just help me to understand one thing: are you sure that component is using smartthings now? If you can select HDMI input / TV means SmarThings working, otherwise is using only local connection.

**Comment by marcoevcom** [comment]
yes, I can change HDMI ports, change Apps, and control play, pause, volume.

**Comment by lkroeker** [comment]
I'm not able to add it from the UI.

I get: "Unable to connect to this Samsung TV device."

Can't see any other issues. According to the documentation adding manually doesn't work anymore? 
Suggestions?

**Comment by mariomazon** [comment]
I have two TV's on my account. The Integration can't recognize my ID, even with right token. When I do a Get to
https://api.smartthings.com/v1/devices with my token as Authorization Bearer (using Postman), I could get the right device ID and set it mannualy on integration Setup. But now, I got a lot of "SamsungTV Smart - Error refreshing from SmartThings" error in my logs - Source: custom_components/samsungtv_smart/media_player.py:665. I can confirm the token AND the device id was right ! Any ideia why this error ?

**Comment by ollo69** [comment]
> I have two TV's on my account. The Integration can't recognize my ID, even with right token. When I do a Get to
> https://api.smartthings.com/v1/devices with my token as Authorization Bearer (using Postman), I could get the right device ID and set it mannualy on integration Setup. But now, I got a lot of "SamsungTV Smart - Error refreshing from SmartThings" error in my logs - Source: custom_components/samsungtv_smart/media_player.py:665. I can confirm the token AND the device id was right ! Any ideia why this error ?

Probably your HA is behind a firewall that is blocking access to SmartThings. For this reason both setup and refresh API calls  fails.
